test/clojupyter/misc/kind_test.clj

```diff
@@ -0,0 +1,152 @@
+            
+            [midje.sweet                    :refer [facts =>]])
+  )
+
+(def raw-image
+  (->  "https://upload.wikimedia.org/wikipedia/commons/e/eb/Ash_Tree_-_geograph.org.uk_-_590710.jpg"
+       (java.net.URL.)
+       (javax.imageio.ImageIO/read))
+  )
+
+(def image
+  (kind/image raw-image
+   ))
+
+(def cs
+  (kind/cytoscape
+   {:elements {:nodes [{:data {:id "a" :parent "b"} :position {:x 215 :y 85}}
+                       {:data {:id "b"}}
+                       {:data {:id "c" :parent "b"} :position {:x 300 :y 85}}
+                       {:data {:id "d"} :position {:x 215 :y 175}}
+                       {:data {:id "e"}}
+                       {:data {:id "f" :parent "e"} :position {:x 300 :y 175}}]
+               :edges [{:data {:id "ad" :source "a" :target "d"}}
+                       {:data {:id "eb" :source "e" :target "b"}}]}
+    :style [{:selector "node"
+             :css {:content "data(id)"
+                   :text-valign "center"
+                   :text-halign "center"}}
+            {:selector "parent"
+             :css {:text-valign "top"
+                   :text-halign "center"}}
+            {:selector "edge"
+             :css {:curve-style "bezier"
+                   :target-arrow-shape "triangle"}}]
+    :layout {:name "preset"
+             :padding 5}}))
+  
+(defn fetch-dataset [dataset-name]  
+  (-> dataset-name
+     (->> (format "https://vincentarelbundock.github.io/Rdatasets/csv/%s.csv"))
+     (tc/dataset {:key-fn (fn [k]
+                            (-> k
+                                str/lower-case
+                                (str/replace #"\." "-")
+                                keyword))})
+     (tc/set-dataset-name dataset-name)))
+
+(def iris
+  (fetch-dataset "datasets/iris"))
+
+
+(facts "eval works for different kinds"
+       (k/kind-eval '(+ 1 1)) => {:html-data "2"}
+
+       (k/kind-eval '(kind/md "# 123")) => {:markdown ["# 123"]}
+
+       (str/starts-with?
+        (-> (k/kind-eval '(kind/image image)) class (.getName))
+        "clojupyter.misc.display$render_mime") => true
+
+       (-> (k/kind-eval '[(kind/image image) (kind/image image)])
+           :html-data
+           (nth 3)
+           (nth 2)
+           (nth 2))
+       => "nested rendering of :kind/image not possible in Clojupyter"
+
+
+       (str/includes?
+        (->
+         (k/kind-eval  '^:kind/cytoscape cs)
+         :html-data
+         (nth 2)
+         second) "cytoscape") => true)
+
+
+(facts "kind/fn works as expected"
+       (str/includes?
+        (->
+         (k/kind-eval  '(-> iris
+                            (plotly/layer-point {:=x :sepal-width
+                                                 :=y :sepal-length
+                                                 :=color :species
+                                                 :=mark-size 10})))
+         :html-data
+         (nth 2)
+         second) "Plotly") => true
+
+       ;; TODO: should work
```

Comment:
Should this be fixed before merging? If not, please delete or adjust appropriately 🙏🏻

Reply:
I kept their several "tests" commented, which ideally should pass one day.
I would call this the difference between 95 % compliant with "kind" and 99 % compliant with kind.
I can remove the todo

Reply:
That's OK - maybe keep the comment but add "making this work would get us closer to 99% compatible with kind"

Reply:
done is last commit